src/main/scala/firrtl/passes/memlib/ReplaceMemMacros.scala

```diff
@@ -24,8 +26,7 @@ object ReplaceMemMacros {
   * This will not generate wmask ports if not needed.
   * Creates the minimum # of black boxes needed by the design.
   */
-class ReplaceMemMacros(writer: ConfWriter) extends Transform with DependencyAPIMigration {
-
+class ReplaceMemMacros extends Transform with DependencyAPIMigration {
```

Comment:
Your deprecation warnings say things are deprecated in 1.4 (to be removed in 1.5), but this is a pretty substantial binary compatibility change. I'm not arguing against the change but it may be tough to do them in a way that is reasonable to backport.

Reply:
I will make these changes 1.5 only and not to backport, after approve, I will start to create another PR to add deprecations o memlib.